### PR TITLE
jupyterhub, jupyter-telemetry: 1.0.0 -> 1.1.0, init at 0.1.0

### PR DESCRIPTION
**Pull request by chiroptical**
Resolve issue with jupyterhub 1.0.0 through upgrade, requires jupyter-telemetry as a dependency

###### Motivation for this change

Experienced an issue with JupyterHub 1.0.0 which is resolved by upgrading to 1.1.0. During the upgrade, determined that jupyter-telemetry is a build dependency.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Unsure how to proceed (errors seem unrelated to change)

- Command `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"` fails with Python XML related error
- Command `nix path-info -S` errors out

### CODE REVIEW

File: pkgs/development/python-modules/jupyter-telemetry/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+, jsonschema
+, ruamel_yaml
+, traitlets
+}:
+
+buildPythonPackage rec {
+  pname = "jupyter_telemetry";
+  version = "0.1.0";
+
+  src = fetchPypi {
```
Comment (jonringer) [inline pkgs/development/python-modules/jupyter-telemetry/default.nix:14]: To comply with [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes) please have the commit message name be of the format
```
<pkg-name>: <subject-line>
```
for more examples, please look at https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes

in your case, the commit message should be a separate commit:
```
pythonPackages.jupyter-telemetry: init at 0.1.0
```

**Review by jonringer — COMMENTED** [review]
should have two commits:
```
pythonPackages.jupyter-telemetry: init at 0.1.0
pythonPackages.jupyterhub: 1.0.0 -> 1.1.0
```

**Comment by jonringer** [comment]
thanks for opening your first PR with nixpkgs :)

File: pkgs/development/python-modules/jupyter-telemetry/default.nix (hunk 1 above)
Comment (chiroptical) [reply pkgs/development/python-modules/jupyter-telemetry/default.nix:14]: Commit messages are fixed.

**Comment by jonringer** [comment]
related: #94783

**Comment by chiroptical** [comment]
> related: #94783

Thanks for pointing this out! This will be huge for where I want to end up.

**Review by jonringer — COMMENTED** [review]
just realized this was made against `release-20.03` please target master branch.

Backports to release branches only really happen if the application is unusable without the backport (E.g. Discord), or security patches.

Full explanation

When backporting changes, please follow https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#backporting-changes.

Namely, you should be doing `git cherry-pick -x <rev>` from a commit that has already landed in master. If the branches have diverged, you may alter the commit or add another commit to ensure that the package is able to still evaluate and build.

File: pkgs/development/python-modules/jupyter-telemetry/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+, fetchPypi
+, python-json-logger
+, jsonschema
+, ruamel_yaml
+, traitlets
+}:
+
+buildPythonPackage rec {
+  pname = "jupyter_telemetry";
+  version = "0.1.0";
```
Comment (jonringer) [inline pkgs/development/python-modules/jupyter-telemetry/default.nix:12]:
```suggestion
  version = "0.1.0";
  disabled = isPy27;
```

File: pkgs/development/python-modules/jupyter-telemetry/default.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
+{ stdenv
+, buildPythonPackage
+, fetchPypi
```
Comment (jonringer) [inline pkgs/development/python-modules/jupyter-telemetry/default.nix:3]:
```suggestion
, fetchPypi
, isPy27
```

**Review by jonringer — COMMENTED** [review]
```
  ERROR: Package 'jupyter-telemetry' requires a different Python: 2.7.18 not in '>=3.5'
  builder for '/nix/store/27242kazi4zb78481lhgxfxvhhd4j590-python2.7-jupyter_telemetry-0.1.0.drv' failed with exit code 1
cannot build derivation '/nix/store/0dsj3sfaaxq1jwpnxgplnj5il1566r7k-env.drv': 1 dependencies couldn't be built
[10 built (1 failed), 271 copied (570.2 MiB), 101.9 MiB DL]
error: build of '/nix/store/0dsj3sfaaxq1jwpnxgplnj5il1566r7k-env.drv' failed
https://github.com/NixOS/nixpkgs/pull/94866
1 package failed to build:
python27Packages.jupyter-telemetry
```

**Comment by chiroptical** [comment]
> just realized this was made against `release-20.03` please target master branch.
> 
> Backports to release branches only really happen if the application is unusable without the backport (E.g. Discord), or security patches.

Can do. I think I misunderstood the documentation when reading it.

**Comment by jonringer** [comment]
@chiroptical you didn't have to close, you could have done:
```
git rebase --onto=origin/master HEAD^^
```
then in github, go to "edit" and change target branch to master

**Comment by chiroptical** [comment]
> ```
> git rebase --onto=origin/master HEAD^^
> git push .. .. --force
> ```

Nice, I didn't realize I could do this. Will fix this up now.

File: pkgs/development/python-modules/jupyter-telemetry/default.nix (hunk 2 above)
Comment (chiroptical) [reply pkgs/development/python-modules/jupyter-telemetry/default.nix:12]: I noticed two things: https://github.com/jupyter/telemetry/blob/341f34421d73f11ec5538da8b04a52eb632f7511/setup.py#L33

and 

https://github.com/NixOS/nixpkgs/blob/84cf00f98031e93f389f1eb93c4a7374a33cc0a9/pkgs/development/python-modules/jupyterhub/default.nix#L57

It seems like these is more appropriate?

**Comment by chiroptical** [comment]
Corrected in #94885, sorry for naivety.

**Comment by jonringer** [comment]
> sorry for naivety.

don't apologize for being unaware. I didn't know either until a few months ago :)
